### PR TITLE
Fix `gc()` bug cause by resource refer count

### PR DESCRIPTION
**Pull request by cptbtptpbcptdtptp**

### CODE REVIEW

File: packages/core/src/2d/sprite/Sprite.ts [hunk 1]
```diff
@@ -264,12 +267,29 @@ export class Sprite extends ReferResource {
     return this._bounds;
   }
 
+  override _addReferCount(value: number): void {
```
Comment (GuoLei1990) [inline packages/core/src/2d/sprite/Sprite.ts:270]: Add `@internal`

File: packages/core/src/shader/ShaderData.ts [hunk 2]
```diff
@@ -613,6 +613,7 @@ export class ShaderData implements IReferable, IClone {
           targetPropertyValueMap[k] = property;
         } else if (property instanceof Texture) {
           targetPropertyValueMap[k] = property;
+          property && property instanceof Texture && referCount > 0 && property._addReferCount(referCount);
```
Comment (GuoLei1990) [inline packages/core/src/shader/ShaderData.ts:616]: This maybe no need?

File: packages/core/src/texture/RenderTarget.ts [hunk 3]
```diff
@@ -167,6 +167,7 @@ export class RenderTarget extends GraphicsResource {
         if (colorTextures[i]._isDepthTexture) {
           throw "Render texture can't use depth format.";
         }
+        colorTextures[i]._addReferCount(1);
```
Comment (GuoLei1990) [inline packages/core/src/texture/RenderTarget.ts:170]: Cache `colorTexture`

Comment (cptbtptpbcptdtptp) [reply packages/core/src/texture/RenderTarget.ts:170]: Done

File: packages/core/src/shader/ShaderData.ts (hunk 2 above)
Comment (cptbtptpbcptdtptp) [reply packages/core/src/shader/ShaderData.ts:616]: Done.

File: packages/core/src/2d/sprite/Sprite.ts (hunk 1 above)
Comment (cptbtptpbcptdtptp) [reply packages/core/src/2d/sprite/Sprite.ts:270]: Done.

**Pull request merged**